### PR TITLE
Support 1 or 3 arg in generate_series() UDTF

### LINKED ISSUE #13615: Support 1 or 3 arg in `generate_series()` UDTF
### Is your feature request related to a problem or challenge?

This is a follow up task for https://github.com/apache/datafusion/pull/13540
Now only 2-argument version is supported
```
> select * from generate_series(1,3);
+-------+
| value |
+-------+
| 1     |
| 2     |
| 3     |
+-------+
```

It can be extended with 'no start' and 'step' support:
```
D select * from generate_series(3);
┌─────────────────┐
│ generate_series │
│      int64      │
├─────────────────┤
│               0 │
│               1 │
│               2 │
│               3 │
└─────────────────┘
D select * from generate_series(3,1,-1);
┌─────────────────┐
│ generate_series │
│      int64      │
├─────────────────┤
│               3 │
│               2 │
│               1 │
└─────────────────┘
```

### Describe the solution you'd like

- Extend the UDTF for 1/3 argument support, following the behavior of `DuckDB` https://duckdb.org/docs/sql/functions/list#generate_series
- Add `sqllogictest`s

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by UBarney**
## Which issue does this PR close?

Closes #13615.

## What changes are included in this PR?
+ Added some args validation to `GenerateSeriesTable`
+ Added `step` field to `GenerateSeriesState`, and added support for generating series with custom step

## Are these changes tested?

sqllogictests for generate_series() UDTF

## Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/functions-table/src/generate_series.rs [hunk 1]
```diff
@@ -22,22 +22,25 @@ use async_trait::async_trait;
...
 use std::fmt;
 use std::sync::Arc;
 
+#[derive(Debug, Clone)]
+enum GenSeriesArgs {
+    ContainsNull,
+    AllNotNullArgs { start: i64, end: i64, step: i64 },
+}
+
 /// Table that generates a series of integers from `start`(inclusive) to `end`(inclusive)
```
Comment (goldmedal) [inline datafusion/functions-table/src/generate_series.rs:39]: I believe the comment should be updated for `step`.

File: datafusion/functions-table/src/generate_series.rs [hunk 2]
```diff
@@ -22,22 +22,25 @@ use async_trait::async_trait;
...
 struct GenerateSeriesTable {
     schema: SchemaRef,
-    // None if input is Null
-    start: Option<i64>,
-    // None if input is Null
-    end: Option<i64>,
+    args: GenSeriesArgs,
 }
 
 /// Table state that generates a series of integers from `start`(inclusive) to `end`(inclusive)
```
Comment (goldmedal) [inline datafusion/functions-table/src/generate_series.rs:46]: Same as above. It should be updated.

**Review by goldmedal — APPROVED** [review]
Thanks @UBarney. Overall looks good to me but some minor comments.
cc @2010YOUY01

**Comment by UBarney** [comment]
> Thanks @UBarney. Overall looks good to me but some minor comments. cc @2010YOUY01

Thanks @goldmedal for reviewing. All comments have been addressed. Please take another look

**Review by goldmedal — APPROVED** [review]

**Comment by goldmedal** [comment]
Thanks @UBarney

**Pull request merged**